cmd/run.go

```diff
@@ -61,7 +64,16 @@ export const main = async (...args: string[]) => {
 
 		runOpts.Args = args[1:]
 
-		err = internal.Run(repo, runOpts)
+		ctx, cancel := context.WithCancel(cmd.Context())
+		shutdown := make(chan os.Signal, 1)
+		signal.Notify(shutdown, syscall.SIGINT)
+		go func() {
+			_ = <-shutdown
+			fmt.Fprintf(os.Stderr, "Received shutdown signal. Waiting for process to finish.\n")
```

Comment:
This is kinda a lie, if we've got a timeout and then kill without waiting. Also, if we're logging this, we probably should log when that timeout fires. If we have both of those logging things, maybe they belong close together in the code?